processor/elasticapmprocessor/internal/routing/data_stream.go

```diff
@@ -125,3 +127,92 @@ func EncodeErrorDataStream(attributes pcommon.Map, dataStreamType string) {
 	attributes.PutStr("data_stream.dataset", "apm.error")
 	attributes.PutStr("data_stream.namespace", NamespaceDefault)
 }
+
+// hasTransactionSpanContext checks if the attributes contain transaction or span context.
+// This is used to identify metrics that include transaction/span fields, which should be
+// routed to internal metrics data streams.
+func hasTransactionSpanContext(attributes pcommon.Map) bool {
+	// Check for transaction-related attributes
+	if _, ok := attributes.Get("transaction.id"); ok {
+		return true
+	}
+	if _, ok := attributes.Get("transaction.name"); ok {
+		return true
+	}
+	if _, ok := attributes.Get("transaction.type"); ok {
+		return true
+	}
+
+	// Check for span-related attributes
+	if _, ok := attributes.Get("span.id"); ok {
+		return true
+	}
+	if _, ok := attributes.Get("span.name"); ok {
+		return true
+	}
+
+	return false
+}
+
+// getMetricsetInterval extracts the metricset.interval attribute value,
+// or empty string if not present.
+func getMetricsetInterval(attributes pcommon.Map) string {
+	if interval, ok := attributes.Get("metricset.interval"); ok {
+		return interval.Str()
+	}
+	return ""
+}
+
+// getMetricsetName extracts the metricset.name attribute value,
+// or empty string if not present.
+func getMetricsetName(attributes pcommon.Map) string {
+	if name, ok := attributes.Get("metricset.name"); ok {
+		return name.Str()
+	}
+	return ""
+}
+
+func isServiceSummary(attributes pcommon.Map) bool {
+	return getMetricsetName(attributes) == "service_summary"
+}
+
+// Using default "apm.internal" data stream
+func internalMetricDataStream(attributes pcommon.Map, dataStreamType string) {
+	attributes.PutStr("data_stream.type", dataStreamType)
+	attributes.PutStr("data_stream.dataset", "apm.internal")
+	attributes.PutStr("data_stream.namespace", NamespaceDefault)
+}
+
+// Data stream formatted as: apm.${metricset.name}.${metricset.interval}
+func internalIntervalMetricDataStream(attributes pcommon.Map, dataStreamType, metricsetName, interval string) {
+	attributes.PutStr("data_stream.type", dataStreamType)
+	attributes.PutStr("data_stream.dataset", fmt.Sprintf("apm.%s.%s", metricsetName, interval))
+	attributes.PutStr("data_stream.namespace", NamespaceDefault)
+}
+
+// EncodeDataStreamMetricDataPoint determines and sets the appropriate data stream for a metric data point.
+// This implements the routing logic from apm-data's metricsetDataset function.
+func EncodeDataStreamMetricDataPoint(attributes pcommon.Map, metricName string, hasServiceName bool) {
```

Comment:
```suggestion
// EncodeDataStreamInternalMetricDataPoint determines if the metric represents an internal metric and sets the appropriate data stream for a metric data point.
// This implements the routing logic from apm-data's metricsetDataset function.
func EncodeDataStreamInternalMetricDataPoint(attributes pcommon.Map, metricName string, hasServiceName bool) {
```

Suggestion to rename update the godoc to specify that the datastream is only set for internal metrics